### PR TITLE
Fix calculation of Settings hash related to Cache commands

### PR DESCRIPTION
**Pull request by natanrolnik**
### Short description 📝

In our projects, every time `tuist cache print-hashes` ran, it would result in different hashes.

### Solution 📦

When debugging Tuist, I noticed that the hash generated by `SettingsContentHasher` was different between runs.
After investigating it a little bit, I found that `SettingsContentHasher` used to receive SettingsDictionary and generate a string based on the keys and values. However, it wasn’t sorted according to the keys, so every run could yield a different hash for the settings.

I also changed the implementation of `hash(_ settingsDictionary: SettingsDictionary)` to match its naming, and return the hash of the settings instead of simply a string containing its contents.

### CODE REVIEW

File: Sources/TuistCache/ContentHashing/DependencyContentHasher.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ public final class DependenciesContentHasher: DependenciesContentHashing {
         self.contentHasher = contentHasher
     }
 
-    // MARK: - HeadersContentHashing
+    // MARK: - DependenciesContentHashing
```
Comment (andreacipriani) [inline Sources/TuistCache/ContentHashing/DependencyContentHasher.swift:19]: copy/paste traces.... :)

**Review by andreacipriani — APPROVED** [review]
LGTM! Thanks for the fix.
Do you think it would be possible to update the unit test to make it fail in case settings were not sorted?

File: Sources/TuistCache/ContentHashing/DependencyContentHasher.swift (hunk 1 above)
Comment (natanrolnik) [reply Sources/TuistCache/ContentHashing/DependencyContentHasher.swift:19]: I imagined 😃 There was another one as well.

**Review by pepibumur — APPROVED** [review]

**Comment by pepibumur** [comment]
@all-contributors add @natanrolnik for bug

**Comment by pepibumur** [comment]
@natanrolnik I think you'll have to update the associated to include the right hash.

**Pull request merged**